src/components/DataFetcher/index.tsx

```diff
@@ -58,6 +60,14 @@ const DataFetcher: React.FC<Props> = ({
    * or if we had an error when calling the API.
    */
   if (error) {
+    // if there is a custom error renderer, use it.
+    if (errorRenderer) {
+      const errorComponent = errorRenderer(error);
+      // if the custom error renderer returns false, it means that it doesn't want to render anything special.
+      if (errorComponent !== false) {
+        return errorComponent;
+      }
```

Comment:
The use case I have is I want to have a custom `errorRenderer` that checks for a specific case only and return a custom error, otherwise, just render the default `return <Error onRetryClicked={onRetryClicked} error={error} />;` bit. This is how it looks like:

```
const renderError = (error: any) => {
    if (error?.message === ApiErrorMessage.CourseNotEnrolled) {
      return (
        <div className={styles.container}>
          <PageContainer>
            <Trans
              i18nKey="learn:not-enrolled"
              components={{
                link: (
                  <Link
                    onClick={onUnEnrolledNavigationLinkClicked}
                    key={0}
                    href={getCourseNavigationUrl(slug as string)}
                    variant={LinkVariant.Blend}
                  />
                ),
              }}
            />
          </PageContainer>
        </div>
      );
    }
    return false;
  };
  ```